torch/_inductor/fx_passes/quantization.py

```diff
@@ -143,8 +143,109 @@ def qconv(match: Match, *args, **kwargs):
+    KeywordArg("dynamic_x_scale"),  # x_scale
+    KeywordArg("dynamic_x_zp"),  # x_zp
+    KeywordArg("packed_weight"),  # packed_weight
+    KeywordArg("w_scale"),  # w_scale
+    KeywordArg("w_zp"),  # w_zp
+    KeywordArg("w_axis"),  # w_axis
+    KeywordArg("b"),  # bias
+    KeywordArg("stride"),
+    KeywordArg("padding"),
+    KeywordArg("dilation"),
+    KeywordArg("transposed"),
+    KeywordArg("o_padding"),
+    KeywordArg("groups"),
+)
+
+quantize_conv_output_pattern_pt2e = CallFunction(
+    prims.convert_element_type.default,
+    CallFunction(
+        aten.clamp_max.default,
+        CallFunction(
+            aten.clamp_min.default,
+            CallFunction(
+                aten.add.Tensor,
+                CallFunction(
+                    aten.round.default,
+                    CallFunction(
+                        aten.mul.Tensor,
+                        aten_qconv_pt2e_pattern,  # output of conv
+                        KeywordArg("o_inv_scale"),
+                    ),
+                ),
+                KeywordArg("o_zp"),
+            ),
+            KeywordArg("o_qmin"),  # 0
+        ),
+        KeywordArg("o_qmax"),  # 127
+    ),
+    KeywordArg("o_dtype"),  # dtype=torch.uint8
+)
+
+pattern_match_count = 0
+
+
+def _register_quantized_conv_lowering_pt2e(pattern):
+    @register_lowering_pattern(pattern)
+    def qconv(match: Match, *args, **kwargs):
+        x, x_scale, x_zp = kwargs["x"], kwargs["x_scale"], kwargs["x_zp"]
+        b, stride, padding, dilation = (
+            kwargs["b"],
+            kwargs["stride"],
+            kwargs["padding"],
+            kwargs["dilation"],
+        )
+        groups, o_inv_scale, o_zero_point, o_dtype = (
+            kwargs["groups"],
+            kwargs["o_inv_scale"],
+            kwargs["o_zp"],
+            kwargs["o_dtype"],
+        )
+
+        # packed_weight = kwargs["packed_weight"]
+        packed_weight, w_scale, w_zp, w_axis = (
+            kwargs["packed_weight"],
+            kwargs["w_scale"],
+            kwargs["w_zp"],
+            kwargs["w_axis"],
+        )
+        global pattern_match_count
+        pattern_match_count += 1
+        print(
+            "---- matched the pattern ----: {}".format(pattern_match_count), flush=True
+        )
+
+        weight_shape = packed_weight.get_size()
+        dim = len(weight_shape) - 2
+        return QConvPT2E.create(
+            dim,
+            x,
+            x_scale,
+            x_zp,
+            packed_weight,
+            w_scale,
+            w_zp,
+            w_axis,
+            b,
+            stride,
+            padding,
+            dilation,
+            groups,
+            o_inv_scale,
+            o_zero_point,
+            o_dtype,
+        )
+
+    return qconv
+
+
 def register_quantization_lowerings():
     _register_quantized_conv_lowering(quantize_conv_output_pattern)
+    _register_quantized_conv_lowering_pt2e(quantize_conv_output_pattern_pt2e)
```

Comment:
why does pt2e pattern have packed weight instead of dequantized weight and then aten.convolution?

Reply:
Also writing these pattern manually also seems to rely on the decomposition rules and how the op will appear in the exported graph. Is that a valid assumption?

Reply:
> why does pt2e pattern have packed weight instead of dequantized weight and then aten.convolution?

Weight prepacking is done earlier in inductor as part of the constant folding pass in freezing graph pass where weights are marked as constants and folded with "prepacking op" into the prepacked weights. We don't create new constant tensors in the graph fusion passes.

> Also writing these pattern manually also seems to rely on the decomposition rules and how the op will appear in the exported graph. Is that a valid assumption?

That's a good question. I guess it is a nature of pattern-based approach - we have to make assumptions about how the graph would look like. I think that should be fine since the decomposition rules are also what PyTorch core maintains.